saladItem.js

```diff
@@ -1,3 +1,68 @@
 import React, { useReducer, useContext } from 'react';
 import PropTypes from 'prop-types';
 import { createUseStyles } from 'react-jss';
+
+import UserContext from '../User/User';
+import { SaladContext } from '../SaladMaker/SaladMaker';
+
+// Items for a list of Salad options for the Salad Maker Interface
+const useStyles = createUseStyles({
+        add: {
+          background: 'none',
+          border: 'none',
+          cursor: 'pointer',
+        },
+            favorite: {
+            fontSize: 20,
+            position: 'absolute',
+            top: 10,
+            right: 10,
+            },
+        image: {
+          fontSize: 80
+        },
+    wrapper: {
+        border: 'lightgrey solid 1px',
+        margin: 20,
+        padding: 25,
+        position: 'relative',
+        textAlign: 'center',
+        textTransform: 'capitalize',
+        width: 200,
+    }
+});
+
+var reducer = 0;
+const reducer = key => key + 1;
+// Provides a viewable list of Salad Items by Image and Name
+export default function SaladItem({ image, name }) {
+  const classes = useStyles();
+  const { setSalad } = useContext(SaladContext)
+  const user = useContext(UserContext);
+  const favorite = user.favorites.includes(name);
+  const [id, updateId] = useReducer(reducer, 0);
+  function update() { 
```

Comment:
- update is an event handler that fires when the button is clicked. It will call the dispatcher function, incrementing the Id by + 1, and update the state of salad via setSalad.